src/Sylius/Behat/Context/Ui/Admin/ManagingPaymentMethodsContext.php

```diff
@@ -0,0 +1,372 @@
+        CreatePageInterface $createPage,
+        IndexPageInterface $indexPage,
+        UpdatePageInterface $updatePage,
+        CurrentPageResolverInterface $currentPageResolver,
+        NotificationCheckerInterface $notificationChecker
+    ) {
+        $this->createPage = $createPage;
+        $this->indexPage = $indexPage;
+        $this->updatePage = $updatePage;
+        $this->currentPageResolver = $currentPageResolver;
+        $this->notificationChecker = $notificationChecker;
+    }
+
+    /**
+     * @Given I want to modify the :paymentMethod payment method
+     */
+    public function iWantToModifyAPaymentMethod(PaymentMethodInterface $paymentMethod)
+    {
+        $this->updatePage->open(['id' => $paymentMethod->getId()]);
+    }
+
+    /**
+     * @When I name it :name in :language
+     * @When I rename it to :name in :language
+     * @When I remove its name from :language translation
+     */
+    public function iNameItIn($name = null, $language)
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm($this->createPage, $this->updatePage);
+
+        $currentPage->nameIt($name, $language);
+    }
+
+    /**
+     * @When I do not name it
+     */
+    public function iDoNotNameIt()
+    {
+        // Intentionally left blank to fulfill context expectation
+    }
+
+    /**
+     * @When I enable it
+     */
+    public function iEnableIt()
+    {
+        $this->updatePage->enable();
+    }
+
+    /**
+     * @When I disable it
+     */
+    public function iDisableIt()
+    {
+        $this->updatePage->disable();
+    }
+
+    /**
+     * @When I save my changes
+     * @When I try to save my changes
+     */
+    public function iSaveMyChanges()
+    {
+        $this->updatePage->saveChanges();
+    }
+
+    /**
+     * @When I delete the :paymentMethod payment method
+     */
+    public function iDeletePaymentMethod(PaymentMethodInterface $paymentMethod)
+    {
+        $this->indexPage->open();
+        $this->indexPage->deleteResourceOnPage(['code' => $paymentMethod->getCode(), 'name' => $paymentMethod->getName()]);
+    }
+
+    /**
+     * @When I choose :gatewayName gateway
+     */
+    public function iChooseGateway($gatewayName)
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm($this->createPage, $this->updatePage);
+
+        $currentPage->chooseGateway($gatewayName);
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully edited
+     */
+    public function iShouldBeNotifiedAboutSuccessfulEdition()
+    {
+        $this->notificationChecker->checkEditionNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then this payment method :element should be :value
+     */
+    public function thisPaymentMethodElementShouldBe($element, $value)
+    {
+        Assert::true(
+            $this->updatePage->hasResourceValues([$element => $value]),
+            sprintf('Payment method should have %s %s', $value, $element)
```

Comment:
IMO it is correct
I guess that `Payment method should have Offline gateway` is more natural than `Payment method should have gateway Offline`